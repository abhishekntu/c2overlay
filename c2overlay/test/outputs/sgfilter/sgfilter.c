/****************************************************************/
/*   chebyshev		                                        */
/****************************************************************/

#ifdef TEST
#include <stdio.h>
#endif

#include<math.h>

#ifdef ALGO
int foo(int x, int y)
{
 	return (x*(x*(7*x - 76*y +7) + y*(92*y - 39) +7) - y*(y*(984*y + 46) + 46) -75);	
}
#endif


#ifdef TEST
int main(void)
{
	int x,y;
 	int result;
  	result = foo(x,y);
  	return 0;
}
#endif
