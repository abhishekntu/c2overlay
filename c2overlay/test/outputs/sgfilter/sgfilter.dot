digraph graphname {
N12 [color="black", label="N7N6_mul_N7_add_N6_N12"];
N13 [color="black", label="O0_N21_N13"];
N10 [color="black", label="N3N8_add_Imm_7_N3_mul_N8_N10"];
N11 [color="black", label="N5N14_add_Imm_7_N5_mul_N14_N11"];
N8 [color="black", label="N15N18_mul_Imm_-984_N15_sub_Imm_46_N18_N8"];
N9 [color="black", label="N9N20_mul_Imm_92_N9_sub_Imm_39_N20_N9"];
N1 [color="black", label="I0_N1_N1"];
N2 [color="black", label="I1_N2_N2"];
N3 [color="black", label="N12N19_mul_N12_sub_Imm_46_N19_N3"];
N4 [color="black", label="N13N16_add_N13_sub_Imm_75_N16_N4"];
N5 [color="black", label="N10N4_mul_Imm_7_N10_add_N4_N5"];
N6 [color="black", label="mul_Imm_-76_N11_N6"];
N7 [color="black", label="mul_N17_N7"];
N12 -> N11;
N10 -> N12;
N11 -> N4;
N8 -> N3;
N9 -> N12;
N1 -> N12;
N1 -> N3;
N1 -> N9;
N1 -> N7;
N1 -> N8;
N1 -> N6;
N2 -> N5;
N2 -> N10;
N2 -> N11;
N3 -> N7;
N4 -> N13;
N5 -> N10;
N6 -> N5;
N7 -> N4;
}
