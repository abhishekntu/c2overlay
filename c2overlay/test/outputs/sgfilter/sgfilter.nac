procedure foo(in s32 x,in s32 y,out s32 D_2404)
{
  localvar s32 D_2405;
  localvar s32 D_2406;
  localvar s32 D_2407;
  localvar s32 D_2408;
  localvar s32 D_2409;
  localvar s32 D_2410;
  localvar s32 D_2411;
  localvar s32 D_2412;
  localvar s32 D_2413;
  localvar s32 D_2414;
  localvar s32 D_2415;
  localvar s32 D_2416;
  localvar s32 D_2417;
  localvar s32 D_2418;
  localvar s32 D_2419;
  localvar s32 D_2420;
  localvar s32 D_2421;
L0005:
  D_2405 <= mul x,7;
  D_2406 <= mul y,-76;
  D_2407 <= add D_2405,D_2406;
  D_2408 <= add D_2407,7;
  D_2409 <= mul D_2408,x;
  D_2410 <= mul y,92;
  D_2411 <= sub D_2410,39;
  D_2412 <= mul D_2411,y;
  D_2413 <= add D_2409,D_2412;
  D_2414 <= add D_2413,7;
  D_2415 <= mul D_2414,x;
  D_2416 <= mul y,-984;
  D_2417 <= sub D_2416,46;
  D_2418 <= mul D_2417,y;
  D_2419 <= sub D_2418,46;
  D_2420 <= mul D_2419,y;
  D_2421 <= add D_2415,D_2420;
  D_2404 <= sub D_2421,75;
}
