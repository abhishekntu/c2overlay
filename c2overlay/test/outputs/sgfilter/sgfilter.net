.input N1
pinlist: N1
    
.input N2
pinlist: N2
    
.output out:N13
pinlist: N13
    
.clb N12_blk
pinlist: N9 N1 N10 open N12 open open open open 
subblock: N12_blk 0 1 2 open 4 open open open open
    
.clb N10_blk
pinlist: N5 N2 open open N10 open open open open 
subblock: N10_blk 0 1 open open 4 open open open open

.clb N11_blk
pinlist: N2 N12 open open N11 open open open open 
subblock: N11_blk 0 1 open open 4 open open open open

.clb N8_blk
pinlist: N1 open open open N8 open open open open 
subblock: N8_blk 0 open open open 4 open open open open

.clb N9_blk
pinlist: N1 open open open N9 open open open open 
subblock: N9_blk 0 open open open 4 open open open open

.clb N3_blk
pinlist: N8 N1 open open N3 open open open open 
subblock: N3_blk 0 1 open open 4 open open open open

.clb N4_blk
pinlist: N7 N11 open open N13 open open open open 
subblock: N4_blk 0 1 open open 4 open open open open

.clb N5_blk
pinlist: N6 N2 open open N5 open open open open 
subblock: N5_blk 0 1 open open 4 open open open open

.clb N6_blk
pinlist: N1 open open open N6 open open open open 
subblock: N6_blk 0 open open open 4 open open open open

.clb N7_blk
pinlist: N3 N1 open open N7 open open open open 
subblock: N7_blk 0 1 open open 4 open open open open

