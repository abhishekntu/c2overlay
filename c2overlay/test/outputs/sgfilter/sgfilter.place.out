Netlist file: test.net   Architecture file: test.arch.xml
Array size: 5 x 5 logic blocks

#block name	x	y	subblk	block number
#----------	--	--	------	------------
N1		6	2	0	#0
N2		6	1	0	#1
out:N13		2	0	0	#2
N12_blk		4	2	0	#3
N10_blk		4	1	0	#4
N11_blk		3	1	0	#5
N8_blk		1	1	0	#6
N9_blk		3	2	0	#7
N3_blk		1	2	0	#8
N4_blk		5	1	0	#9
N5_blk		5	2	0	#10
N6_blk		2	1	0	#11
N7_blk		2	2	0	#12
